system-tests/runtimes/file-transfer-provider/src/main/java/org/eclipse/dataspaceconnector/extensions/api/FileTransferDataSink.java

```diff
@@ -0,0 +1,79 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.extensions.api;
+
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.DataSource;
+import org.eclipse.dataspaceconnector.dataplane.spi.pipeline.ParallelSink;
+import org.eclipse.dataspaceconnector.dataplane.spi.result.TransferResult;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.util.List;
+import java.util.Objects;
+
+import static java.lang.String.format;
+import static org.eclipse.dataspaceconnector.spi.response.ResponseStatus.ERROR_RETRY;
+
+class FileTransferDataSink extends ParallelSink {
+    private File file;
+
+    @Override
+    protected TransferResult transferParts(List<DataSource.Part> parts) {
+        for (DataSource.Part part : parts) {
```

Comment:
If I understand correctly, currently only 1 part is supported (as there is only one output file). Should we check this here and process the single part instead?

Reply:
We could, but this hardly seems like a responsibility for the Sink, given its contract?